pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1009,17 +1057,15 @@ private CompletableFuture<? extends Subscription> getNonDurableSubscription(Stri
                 }
 
                 Position startPosition = new PositionImpl(ledgerId, entryId);
-                ManagedCursor cursor = null;
                 try {
-                    cursor = ledger.newNonDurableCursor(startPosition, subscriptionName, initialPosition,
-                            isReadCompacted);
+                    final ManagedCursor cursor = ledger.newNonDurableCursor(startPosition, subscriptionName,
+                            initialPosition, isReadCompacted);
+                    subscriptions.computeIfAbsent(subscriptionName, k ->
```

Comment:
Does it look like `subscriptions.computeIfAbsent` has changed the previous logic?

Reply:
> Does it look like subscriptions.computeIfAbsent has changed the previous logic?

No. the original logic is `synchronized: put if not exists.`

Reply:
I feel it is not a necessary change, so I reverted it